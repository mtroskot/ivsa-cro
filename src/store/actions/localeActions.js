import { localeActionTypes } from 'src/constants/actionTypes';

export function changeLocale(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE,
    payload: { locale }
  };
}
export function changeLocaleSuccess(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE_SUCCESS,
    payload: { locale }
  };
}
