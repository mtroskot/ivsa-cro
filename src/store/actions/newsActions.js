import { newsActionTypes } from 'src/constants/actionTypes';

export function fetchNews(refreshing = false) {
  return {
    type: newsActionTypes.FETCH_NEWS,
    payload: { refreshing }
  };
}
export function fetchNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.FETCH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
export function fetchNewsError(error) {
  return {
    type: newsActionTypes.FETCH_NEWS_ERROR,
    payload: { error }
  };
}

export function publishNews(date, publishedBy, notice) {
  return {
    type: newsActionTypes.PUBLISH_NEWS,
    payload: { date, publishedBy, notice }
  };
}

export function publishNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.PUBLISH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function updateNews(newsId, date, publishedBy, notice) {
  return {
    type: newsActionTypes.UPDATE_NEWS,
    payload: { newsId, date, publishedBy, notice }
  };
}

export function updateNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.UPDATE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function deleteNews(newsId) {
  return {
    type: newsActionTypes.DELETE_NEWS,
    payload: { newsId }
  };
}

export function deleteNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.DELETE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
