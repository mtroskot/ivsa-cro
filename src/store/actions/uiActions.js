import { uiActionTypes } from 'src/constants/actionTypes';
import { ObjectUtils } from 'src/utils';

export const startAction = (name, params) => {
  if (!ObjectUtils.isObject(params)) {
    throw Error(`type of ${typeof params} is not an object`);
  }
  return {
    type: uiActionTypes.START_ACTION,
    payload: {
      action: {
        name,
        params
      }
    }
  };
};

export const stopAction = name => ({
  type: uiActionTypes.STOP_ACTION,
  payload: { name }
});

export const refreshActionStart = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_START,
  payload: { refreshAction }
});

export const refreshActionStop = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_STOP,
  payload: { refreshAction }
});

export const togglePopupMessage = (message, position) => ({
  type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
  payload: {
    message,
    position
  }
});
