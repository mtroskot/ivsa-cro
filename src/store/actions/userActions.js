import { userActionTypes } from 'src/constants/actionTypes';

export function authenticateUser(email, password) {
  return {
    type: userActionTypes.AUTH_USER,
    payload: { email, password }
  };
}

export const authenticateUserSuccess = (displayName, email, idToken, expiryDate) => {
  return {
    type: userActionTypes.AUTH_USER_SUCCESS,
    payload: { displayName, email, idToken, expiryDate }
  };
};

export const authenticateUserError = error => {
  return {
    type: userActionTypes.AUTH_USER_ERROR,
    payload: { error }
  };
};

export const logout = () => {
  return {
    type: userActionTypes.LOG_OUT,
    payload: {}
  };
};
