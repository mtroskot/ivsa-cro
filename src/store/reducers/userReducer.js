import { userActionTypes } from 'src/constants/actionTypes';

export const initialState = {
  userData: {
    isAuthenticated: false,
    displayName: null,
    email: null,
    idToken: null,
    expiryDate: null
  },
  error: null
};

const userReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case userActionTypes.AUTH_USER_SUCCESS:
      const { displayName, email, idToken, expiryDate } = payload;
      return {
        ...state,
        userData: {
          isAuthenticated: true,
          displayName,
          email,
          idToken,
          expiryDate
        },
        error: null
      };
    case userActionTypes.AUTH_USER_ERROR:
      return {
        ...state,
        error: payload.error
      };
    case userActionTypes.LOG_OUT:
      return initialState;
    default:
      return state;
  }
};

export default userReducer;
