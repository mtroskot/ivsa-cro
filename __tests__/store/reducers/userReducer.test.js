import userReducer, { initialState } from 'src/store/reducers/userReducer';
import * as userActions from 'src/store/actions/userActions';

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(userReducer(beforeActionState, action)).toEqual(expectedState);
    expect(userReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns authenticateUserSuccess state', () => {
    const displayName = 'displayName';
    const email = 'email';
    const idToken = 'idToken';
    const expiryDate = 'expiryDate';
    const startAction = userActions.authenticateUserSuccess(displayName, email, idToken, expiryDate);
    const expectedState = {
      userData: {
        isAuthenticated: true,
        displayName,
        email,
        idToken,
        expiryDate
      },
      error: null
    };
    expect(userReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns authenticateUserError state', () => {
    const error = 'error';
    const stopAction = userActions.authenticateUserError(error);
    const expectedState = {
      userData: {
        isAuthenticated: false,
        displayName: null,
        email: null,
        idToken: null,
        expiryDate: null
      },
      error
    };
    expect(userReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('4returns logout state', () => {
    const refreshActionStart = userActions.logout();
    const expectedState = initialState;
    expect(userReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(userReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });
});

describe('AUTH_USER_SUCCESS_STATE', () => {
  const beforeActionState = {
    ...initialState,
    userData: {
      isAuthenticated: true,
      displayName: 'displayName',
      email: 'email',
      idToken: 'idToken',
      expiryDate: 'expiryDate'
    }
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(userReducer(beforeActionState, action)).toEqual(expectedState);
    expect(userReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns authenticateUserSuccess state', () => {
    const displayName = 'displayName2';
    const email = 'email2';
    const idToken = 'idToken2';
    const expiryDate = 'expiryDate2';
    const startAction = userActions.authenticateUserSuccess(displayName, email, idToken, expiryDate);
    const expectedState = {
      ...beforeActionState,
      userData: {
        isAuthenticated: true,
        displayName,
        email,
        idToken,
        expiryDate
      }
    };
    expect(userReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns authenticateUserError state', () => {
    const error = 'error';
    const stopAction = userActions.authenticateUserError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(userReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('4returns logout state', () => {
    const refreshActionStart = userActions.logout();
    const expectedState = initialState;
    expect(userReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(userReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });
});

describe('AUTH_USER_ERROR_STATE', () => {
  const beforeActionState = {
    ...initialState,
    error: 'error'
  };

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(userReducer(beforeActionState, action)).toEqual(expectedState);
    expect(userReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns authenticateUserSuccess state', () => {
    const displayName = 'displayName2';
    const email = 'email2';
    const idToken = 'idToken2';
    const expiryDate = 'expiryDate2';
    const startAction = userActions.authenticateUserSuccess(displayName, email, idToken, expiryDate);
    const expectedState = {
      ...beforeActionState,
      userData: {
        isAuthenticated: true,
        displayName,
        email,
        idToken,
        expiryDate
      },
      error: null
    };
    expect(userReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, startAction)).toMatchSnapshot();
  });

  it('3returns authenticateUserError state', () => {
    const error = 'error2';
    const stopAction = userActions.authenticateUserError(error);
    const expectedState = {
      ...beforeActionState,
      error
    };
    expect(userReducer(beforeActionState, stopAction)).toEqual(expectedState);
    expect(userReducer(beforeActionState, stopAction)).toMatchSnapshot();
  });

  it('4returns logout state', () => {
    const refreshActionStart = userActions.logout();
    const expectedState = initialState;
    expect(userReducer(beforeActionState, refreshActionStart)).toEqual(expectedState);
    expect(userReducer(beforeActionState, refreshActionStart)).toMatchSnapshot();
  });
});
