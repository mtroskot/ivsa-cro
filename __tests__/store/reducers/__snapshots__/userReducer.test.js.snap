// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AUTH_USER_ERROR_STATE 1returns initial state 1`] = `
Object {
  "error": "error",
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`AUTH_USER_ERROR_STATE 2returns authenticateUserSuccess state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": "displayName2",
    "email": "email2",
    "expiryDate": "expiryDate2",
    "idToken": "idToken2",
    "isAuthenticated": true,
  },
}
`;

exports[`AUTH_USER_ERROR_STATE 3returns authenticateUserError state 1`] = `
Object {
  "error": "error2",
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`AUTH_USER_ERROR_STATE 4returns logout state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`AUTH_USER_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": "displayName",
    "email": "email",
    "expiryDate": "expiryDate",
    "idToken": "idToken",
    "isAuthenticated": true,
  },
}
`;

exports[`AUTH_USER_SUCCESS_STATE 2returns authenticateUserSuccess state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": "displayName2",
    "email": "email2",
    "expiryDate": "expiryDate2",
    "idToken": "idToken2",
    "isAuthenticated": true,
  },
}
`;

exports[`AUTH_USER_SUCCESS_STATE 3returns authenticateUserError state 1`] = `
Object {
  "error": "error",
  "userData": Object {
    "displayName": "displayName",
    "email": "email",
    "expiryDate": "expiryDate",
    "idToken": "idToken",
    "isAuthenticated": true,
  },
}
`;

exports[`AUTH_USER_SUCCESS_STATE 4returns logout state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`INITIAL_STATE 2returns authenticateUserSuccess state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": "displayName",
    "email": "email",
    "expiryDate": "expiryDate",
    "idToken": "idToken",
    "isAuthenticated": true,
  },
}
`;

exports[`INITIAL_STATE 3returns authenticateUserError state 1`] = `
Object {
  "error": "error",
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;

exports[`INITIAL_STATE 4returns logout state 1`] = `
Object {
  "error": null,
  "userData": Object {
    "displayName": null,
    "email": null,
    "expiryDate": null,
    "idToken": null,
    "isAuthenticated": false,
  },
}
`;
